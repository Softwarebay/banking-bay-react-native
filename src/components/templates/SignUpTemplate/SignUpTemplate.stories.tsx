import { action } from '@storybook/addon-actions';
import { text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';
import { ViewStory } from 'utils/storybook/ViewStory';

import { SignUpTemplate } from './SignUpTemplate';

storiesOf('Templates/SignUpTemplate', module)
	.addDecorator((getStory) => <ViewStory fullscreen>{getStory()}</ViewStory>)
	.add('Default', () => (
		<SignUpTemplate
			footerLinkText={text('footerLinkText', 'Press me!')}
			footerText={text('footerText', 'Footer text.')}
			headingText={text('headingText', 'Main Heading')}
			onPressFacebook={action('onPressFacebook')}
			onPressFooterLink={action('onPressFooterLink')}
			onPressGooglePlus={action('onPressGooglePlus')}
			onPressTwitter={action('onPressTwitter')}
			onSubmitSignUpForm={action('onSubmitSignUpForm')}
		/>
	));
