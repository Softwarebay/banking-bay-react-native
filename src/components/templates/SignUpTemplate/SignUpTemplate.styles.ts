import { StyleSheet } from 'react-native';
import { colors } from 'theme';

export const styles = StyleSheet.create({
	container: {
		backgroundColor: colors.screenBackground,
		flex: 1,
		justifyContent: 'center',
		paddingHorizontal: 20,
	},
	footer: {
		marginBottom: 50,
		marginTop: 30,
	},
	heading: {
		marginBottom: 30,
		paddingHorizontal: 35,
		textAlign: 'center',
	},
});
