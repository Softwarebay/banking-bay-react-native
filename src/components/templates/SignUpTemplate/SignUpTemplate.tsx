import { Flex, Heading, Link, Text } from 'components/atoms';
import { AuthViaSocials, SignUpForm } from 'components/organisms';
import { StatusBar } from 'expo-status-bar';
import { Keyboard, TouchableWithoutFeedback, View } from 'react-native';
import { colors } from 'theme';

import { styles } from './SignUpTemplate.styles';

export interface Props {
	headingText: string;
	footerText: string;
	footerLinkText: string;
	onSubmitSignUpForm: (userCredentials: {
		name: string;
		email: string;
		password: string;
	}) => void;
	onPressFooterLink: () => void;
	onPressFacebook: () => void;
	onPressGooglePlus: () => void;
	onPressTwitter: () => void;
}

export const SignUpTemplate = ({
	headingText,
	footerText,
	footerLinkText,
	onSubmitSignUpForm,
	onPressFacebook,
	onPressGooglePlus,
	onPressTwitter,
	onPressFooterLink,
}: Props) => {
	return (
		<>
			<StatusBar backgroundColor={colors.screenBackground} style="dark" />
			<TouchableWithoutFeedback onPress={Keyboard.dismiss}>
				<View style={styles.container}>
					<Heading
						size="h1"
						style={styles.heading}
						testID="sign-up-heading-test-id"
						text={headingText}
					/>
					<SignUpForm onSubmitSignUpForm={onSubmitSignUpForm} />
					<Flex direction="row" style={styles.footer}>
						<Text
							testID="sign-up-footer-text-test-id"
							text={`${footerText} `}
						/>
						<Link
							testID="sign-up-footer-link-test-id"
							text={footerLinkText}
							onPress={onPressFooterLink}
						/>
					</Flex>
					<AuthViaSocials
						onPressFacebook={onPressFacebook}
						onPressGooglePlus={onPressGooglePlus}
						onPressTwitter={onPressTwitter}
					/>
				</View>
			</TouchableWithoutFeedback>
		</>
	);
};
