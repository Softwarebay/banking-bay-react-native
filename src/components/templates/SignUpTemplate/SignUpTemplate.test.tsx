import { fireEvent, render, screen } from '@testing-library/react-native';
import testRenderer from 'react-test-renderer';

import { SignUpTemplate } from './SignUpTemplate';

import type { Props } from './SignUpTemplate';

describe('SignUpTemplate', () => {
	const onSubmitSignUpForm = jest.fn<
		Props['onSubmitSignUpForm'],
		[{ name: string; email: string; password: string }]
	>();
	const onPressFooterLink = jest.fn<Props['onPressFooterLink'], []>();
	const onPressFacebook = jest.fn<Props['onPressFacebook'], []>();
	const onPressGooglePlus = jest.fn<Props['onPressGooglePlus'], []>();
	const onPressTwitter = jest.fn<Props['onPressTwitter'], []>();

	const props: Props = {
		footerText: 'Footer Text',
		footerLinkText: 'Footer Link',
		headingText: 'Heading',
		onSubmitSignUpForm,
		onPressFooterLink,
		onPressFacebook,
		onPressGooglePlus,
		onPressTwitter,
	};

	it('should match snapshot', () => {
		const tree = testRenderer.create(<SignUpTemplate {...props} />).toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should render sign in heading', () => {
		render(<SignUpTemplate {...props} />);

		const heading = screen.getByTestId('sign-up-heading-test-id');

		expect(heading).toHaveTextContent('Heading');
	});

	it('should render sign up form', () => {
		render(<SignUpTemplate {...props} />);

		const nameInput = screen.getByTestId('sign-up-form-name-input-test-id');
		const emailInput = screen.getByTestId('sign-up-form-email-input-test-id');
		const passwordInput = screen.getByTestId(
			'sign-up-form-password-input-test-id'
		);
		const confirmPasswordInput = screen.getByTestId(
			'sign-up-form-confirm-password-input-test-id'
		);
		const submitBtn = screen.getByTestId('sign-up-form-submit-button-test-id');

		expect(nameInput).not.toBeNull();
		expect(emailInput).not.toBeNull();
		expect(passwordInput).not.toBeNull();
		expect(confirmPasswordInput).not.toBeNull();
		expect(submitBtn).toHaveTextContent('Sign Up');
	});
	it('should render with initially filled input fields', () => {
		render(<SignUpTemplate {...props} />);

		const nameInputField = screen.getByTestId(
			'input-field-sign-up-form-name-input-test-id'
		);
		const emailInputField = screen.getByTestId(
			'input-field-sign-up-form-email-input-test-id'
		);
		const passwordInputField = screen.getByTestId(
			'input-field-sign-up-form-password-input-test-id'
		);
		const confirmPasswordInputField = screen.getByTestId(
			'input-field-sign-up-form-confirm-password-input-test-id'
		);

		expect(nameInputField.props).toHaveProperty('value', 'John Smith');
		expect(emailInputField.props).toHaveProperty('value', 'test@test.com');
		expect(passwordInputField.props).toHaveProperty('value', 'password');
		expect(confirmPasswordInputField.props).toHaveProperty('value', 'password');
	});

	it('should submit form when input fields are filled and email is valid and passwords match', () => {
		render(<SignUpTemplate {...props} />);

		const submitBtn = screen.getByTestId('sign-up-form-submit-button-test-id');

		fireEvent.press(submitBtn);

		expect(onSubmitSignUpForm).toHaveBeenCalledWith(
			expect.objectContaining({
				name: 'John Smith',
				email: 'test@test.com',
				password: 'password',
			})
		);
		expect(onSubmitSignUpForm).toHaveBeenCalledTimes(1);
	});

	it("shouldn't submit form with empty name input", () => {
		render(<SignUpTemplate {...props} />);

		const nameInputField = screen.getByTestId(
			'input-field-sign-up-form-name-input-test-id'
		);
		const submitBtn = screen.getByTestId('sign-up-form-submit-button-test-id');

		fireEvent.changeText(nameInputField, '');
		fireEvent.press(submitBtn);

		expect(onSubmitSignUpForm).not.toHaveBeenCalled();
	});

	it("shouldn't submit form with empty email input", () => {
		render(<SignUpTemplate {...props} />);

		const emailInputField = screen.getByTestId(
			'input-field-sign-up-form-email-input-test-id'
		);
		const submitBtn = screen.getByTestId('sign-up-form-submit-button-test-id');

		fireEvent.changeText(emailInputField, '');
		fireEvent.press(submitBtn);

		expect(onSubmitSignUpForm).not.toHaveBeenCalled();
	});

	it("shouldn't submit form with invalid email input", () => {
		render(<SignUpTemplate {...props} />);

		const emailInputField = screen.getByTestId(
			'input-field-sign-up-form-email-input-test-id'
		);
		const submitBtn = screen.getByTestId('sign-up-form-submit-button-test-id');

		fireEvent.changeText(emailInputField, 'invalidemail');
		fireEvent.press(submitBtn);

		expect(onSubmitSignUpForm).not.toHaveBeenCalled();
	});

	it("shouldn't submit form with empty password inputs", () => {
		render(<SignUpTemplate {...props} />);

		const passwordInputField = screen.getByTestId(
			'input-field-sign-up-form-password-input-test-id'
		);
		const confirmPasswordInputField = screen.getByTestId(
			'input-field-sign-up-form-confirm-password-input-test-id'
		);
		const submitBtn = screen.getByTestId('sign-up-form-submit-button-test-id');

		fireEvent.changeText(passwordInputField, '');
		fireEvent.changeText(confirmPasswordInputField, '');
		fireEvent.press(submitBtn);

		expect(onSubmitSignUpForm).not.toHaveBeenCalled();
	});

	it("shouldn't submit form with mismatched passwords", () => {
		render(<SignUpTemplate {...props} />);

		const passwordInputField = screen.getByTestId(
			'input-field-sign-up-form-password-input-test-id'
		);
		const confirmPasswordInputField = screen.getByTestId(
			'input-field-sign-up-form-confirm-password-input-test-id'
		);
		const submitBtn = screen.getByTestId('sign-up-form-submit-button-test-id');

		fireEvent.changeText(passwordInputField, 'password');
		fireEvent.changeText(confirmPasswordInputField, 'drowssap');
		fireEvent.press(submitBtn);

		expect(onSubmitSignUpForm).not.toHaveBeenCalled();
	});

	it('should display form footer text, form footer link', () => {
		render(<SignUpTemplate {...props} />);

		const footerText = screen.getByTestId('sign-up-footer-text-test-id');
		const footerLink = screen.getByTestId('sign-up-footer-link-test-id');

		expect(footerText).toHaveTextContent('Footer Text');
		expect(footerLink).toHaveTextContent('Footer Link');
	});

	it('should handle press events on footer link button', () => {
		render(<SignUpTemplate {...props} />);

		const footerLink = screen.getByTestId('sign-up-footer-link-test-id');

		fireEvent.press(footerLink);

		expect(onPressFooterLink).toHaveBeenCalledTimes(1);
	});

	it('should render 3 icon buttons to authenticate with social media', () => {
		render(<SignUpTemplate {...props} />);

		const authButtons = screen.getAllByTestId(
			/(facebook|twitter|googlePlus)-icon-btn-test-id/gi
		);

		expect(authButtons).toHaveLength(3);
	});

	it('should handle press events on Facebook icon', () => {
		render(<SignUpTemplate {...props} />);

		const facebookIconBtn = screen.getByTestId(/facebook-icon-btn-test-id/i);

		fireEvent.press(facebookIconBtn);

		expect(onPressFacebook).toHaveBeenCalledTimes(1);
		expect(onPressGooglePlus).not.toHaveBeenCalled();
		expect(onPressTwitter).not.toHaveBeenCalled();
	});

	it('should handle press events on Google+ icon', () => {
		render(<SignUpTemplate {...props} />);

		const googlePlusIconBtn = screen.getByTestId(
			/googlePlus-icon-btn-test-id/i
		);

		fireEvent.press(googlePlusIconBtn);

		expect(onPressGooglePlus).toHaveBeenCalledTimes(1);
		expect(onPressFacebook).not.toHaveBeenCalled();
		expect(onPressTwitter).not.toHaveBeenCalled();
	});

	it('should handle press events on Twitter icon', () => {
		render(<SignUpTemplate {...props} />);

		const twitterIconBtn = screen.getByTestId(/twitter-icon-btn-test-id/i);

		fireEvent.press(twitterIconBtn);

		expect(onPressTwitter).toHaveBeenCalledTimes(1);
		expect(onPressFacebook).not.toHaveBeenCalled();
		expect(onPressGooglePlus).not.toHaveBeenCalled();
	});
});
