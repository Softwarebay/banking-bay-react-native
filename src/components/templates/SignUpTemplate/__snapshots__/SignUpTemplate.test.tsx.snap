// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpTemplate should match snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#FAFCFF",
      "flex": 1,
      "justifyContent": "center",
      "paddingHorizontal": 20,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#1B1D4D",
        },
        Object {
          "fontFamily": "MulishBold",
          "fontSize": 46,
          "lineHeight": 50.5,
        },
        Object {
          "marginBottom": 30,
          "paddingHorizontal": 35,
          "textAlign": "center",
        },
      ]
    }
    testID="sign-up-heading-test-id"
  >
    Heading
  </Text>
  <View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
          },
          Object {
            "flexDirection": "row",
          },
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 10,
              "shadowColor": "#161966",
              "shadowOffset": Object {
                "height": 6,
                "width": 0,
              },
              "shadowOpacity": 0.08,
              "shadowRadius": 16,
            },
            Array [
              Object {
                "height": 55,
                "paddingHorizontal": 20,
                "paddingVertical": 14,
              },
              Object {
                "marginBottom": 14,
              },
            ],
          ],
        ]
      }
      testID="sign-up-form-name-input-test-id"
    >
      <TextInput
        autoCorrect={false}
        keyboardType="default"
        onChangeText={[Function]}
        placeholder="Enter your name"
        secureTextEntry={false}
        style={
          Array [
            Object {
              "color": "#1B1D4D",
              "fontFamily": "MulishRegular",
              "fontSize": 15,
            },
            Object {
              "flex": 1,
            },
          ]
        }
        testID="input-field-sign-up-form-name-input-test-id"
        value=""
      />
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
          },
          Object {
            "flexDirection": "row",
          },
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 10,
              "shadowColor": "#161966",
              "shadowOffset": Object {
                "height": 6,
                "width": 0,
              },
              "shadowOpacity": 0.08,
              "shadowRadius": 16,
            },
            Array [
              Object {
                "height": 55,
                "paddingHorizontal": 20,
                "paddingVertical": 14,
              },
              Object {
                "marginBottom": 14,
              },
            ],
          ],
        ]
      }
      testID="sign-up-form-email-input-test-id"
    >
      <TextInput
        autoCorrect={false}
        keyboardType="email-address"
        onChangeText={[Function]}
        placeholder="Enter your email"
        secureTextEntry={false}
        style={
          Array [
            Object {
              "color": "#1B1D4D",
              "fontFamily": "MulishRegular",
              "fontSize": 15,
            },
            Object {
              "flex": 1,
              "marginRight": 20,
            },
          ]
        }
        testID="input-field-sign-up-form-email-input-test-id"
        value=""
      />
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
          },
          Object {
            "flexDirection": "row",
          },
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 10,
              "shadowColor": "#161966",
              "shadowOffset": Object {
                "height": 6,
                "width": 0,
              },
              "shadowOpacity": 0.08,
              "shadowRadius": 16,
            },
            Array [
              Object {
                "height": 55,
                "paddingHorizontal": 20,
                "paddingVertical": 14,
              },
              Object {
                "marginBottom": 14,
              },
            ],
          ],
        ]
      }
      testID="sign-up-form-password-input-test-id"
    >
      <TextInput
        autoCorrect={false}
        keyboardType="default"
        onChangeText={[Function]}
        placeholder="Enter your password"
        secureTextEntry={true}
        style={
          Array [
            Object {
              "color": "#1B1D4D",
              "fontFamily": "MulishRegular",
              "fontSize": 15,
            },
            Object {
              "flex": 1,
              "marginRight": 20,
            },
          ]
        }
        testID="input-field-sign-up-form-password-input-test-id"
        value=""
      />
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        hitSlop={20}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Object {
              "right": 0,
            },
            false,
          ]
        }
        testID="icon-btn-sign-up-form-password-input-test-id"
      >
        <SvgMock
          color="#1B1D4D"
          height={16}
          testID="eye-off-icon-test-id"
          width={16}
        />
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
          },
          Object {
            "flexDirection": "row",
          },
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 10,
              "shadowColor": "#161966",
              "shadowOffset": Object {
                "height": 6,
                "width": 0,
              },
              "shadowOpacity": 0.08,
              "shadowRadius": 16,
            },
            Array [
              Object {
                "height": 55,
                "paddingHorizontal": 20,
                "paddingVertical": 14,
              },
              Object {
                "marginBottom": 14,
              },
            ],
          ],
        ]
      }
      testID="sign-up-form-confirm-password-input-test-id"
    >
      <TextInput
        autoCorrect={false}
        keyboardType="default"
        onChangeText={[Function]}
        placeholder="Confirm your password"
        secureTextEntry={true}
        style={
          Array [
            Object {
              "color": "#1B1D4D",
              "fontFamily": "MulishRegular",
              "fontSize": 15,
            },
            Object {
              "flex": 1,
              "marginRight": 20,
            },
          ]
        }
        testID="input-field-sign-up-form-confirm-password-input-test-id"
        value=""
      />
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        hitSlop={20}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Object {
              "right": 0,
            },
            false,
          ]
        }
        testID="icon-btn-sign-up-form-confirm-password-input-test-id"
      >
        <SvgMock
          color="#1B1D4D"
          height={16}
          testID="eye-off-icon-test-id"
          width={16}
        />
      </View>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          false,
          undefined,
        ]
      }
      testID="sign-up-form-submit-button-test-id"
    >
      <ViewManagerAdapter_ExpoLinearGradient
        proxiedProperties={
          Object {
            "colors": Array [
              4288142079,
              4279654301,
            ],
            "endPoint": Array [
              1,
              1,
            ],
            "locations": undefined,
            "startPoint": Array [
              0,
              0,
            ],
          }
        }
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 10,
            "height": 55,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontFamily": "MulishSemiBold",
                "fontSize": 17,
              },
              Object {
                "color": "#FFFFFF",
              },
            ]
          }
        >
          Sign Up
        </Text>
      </ViewManagerAdapter_ExpoLinearGradient>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
        },
        Object {
          "flexDirection": "row",
        },
        Object {
          "marginBottom": 50,
          "marginTop": 30,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#4C4C60",
            "fontFamily": "MulishRegular",
            "fontSize": 17,
            "lineHeight": 27.2,
          },
          undefined,
        ]
      }
      testID="sign-up-footer-text-test-id"
    >
      Footer Text 
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={20}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={false}
      testID="sign-up-footer-link-test-id"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FF5887",
              "fontFamily": "MulishMedium",
              "fontSize": 17,
              "lineHeight": 27.2,
            },
            undefined,
          ]
        }
      >
        Footer Link
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
        },
        Object {
          "flexDirection": "row",
        },
        Object {
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={20}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
          Array [
            Object {
              "borderRadius": 100,
              "height": 60,
              "marginLeft": 10,
              "width": 60,
            },
            Object {
              "backgroundColor": "#3B5999",
            },
          ],
          false,
        ]
      }
      testID="Facebook-icon-btn-test-id"
    >
      <SvgMock
        color="#FFFFFF"
        height={27}
        testID="facebook-icon-test-id"
        width={27}
      />
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={20}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
          Array [
            Object {
              "borderRadius": 100,
              "height": 60,
              "marginLeft": 10,
              "width": 60,
            },
            Object {
              "backgroundColor": "#55ACEE",
            },
          ],
          false,
        ]
      }
      testID="Twitter-icon-btn-test-id"
    >
      <SvgMock
        color="#FFFFFF"
        height={27}
        testID="twitter-icon-test-id"
        width={27}
      />
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={20}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
          Array [
            Object {
              "borderRadius": 100,
              "height": 60,
              "marginLeft": 10,
              "width": 60,
            },
            Object {
              "backgroundColor": "#DD4B39",
            },
          ],
          false,
        ]
      }
      testID="GooglePlus-icon-btn-test-id"
    >
      <SvgMock
        color="#FFFFFF"
        height={27}
        testID="google-plus-icon-test-id"
        width={27}
      />
    </View>
  </View>
</View>
`;
